src/uploadNotionImagesToCloudinary.ts

```diff
@@ -71,11 +72,22 @@ export default async function uploadNotionImagesToCloudinary({
     ////////////////////
     // cover
     ////////////////////
-    const coverUrl =
+    let coverUrl =
       "cover" in page && page.cover?.type === "file"
         ? page.cover.file.url
         : undefined;
 
+    // case if external image is not stored on cloudinary
+    if (
+      uploadExternalsNotOnCloudinary &&
+      "cover" in page &&
+      page.cover?.type === "external" &&
+      !page.cover.external.url.includes("cloudinary")
+    ) {
+      log.debug(`${page.id}: coverUrl not on cloudinary`, page.cover.external)
+      coverUrl = page.cover.external.url
+    }
```

Comment:
Just a thought: 

There's so much similarity between the conditions for cover, icon and images that it might be nice to abstract it a bit. Not necessarily to do right now, could be done later in the future. Something like this:

```js
const coverUrl = getImageUrlToUpload({
  image: "cover" in page ? page.cover : undefined,
  uploadExternalsNotOnCloudinary,
});

function getImageUrlToUpload({ image, uploadExternalsNotOnCloudinary }) {
  if (!image) {
    return undefined;
  }
  // hosted in Notion, then we want to copy it to Cloudinary
  if (image.type === "file") {
    return image.file.url;
  }
  // Hosted externally, but we still want to upload it to Cloudinary
  if (
      uploadExternalsNotOnCloudinary &&
      image.type === "external" &&
      !image.external.url.includes("cloudinary")
  ) {
    return image.external.url;
  }
  return undefined;
}
```

Reply:
good idea 😉